packages/app-evm/src/operations/signTxn/index.ts

```diff
@@ -54,6 +61,24 @@ export const signTxn = async (
 
   await configureAppId(sdk, Number(decodedTxn.chainId));
 
+  const version = await sdk.getAppVersion(
+    getAppIdFromDerivationPaths(Number(decodedTxn.chainId)),
+  );
+  if (
+    version &&
+    compareVersions(
+      `${version.major}.${version.minor}.${version.patch}`,
+      APP_SUPPORT_EIP1559,
+    ) === -1
+  ) {
+    // evm apps support eip2930/eip1559 transactions version 1.1.0 onwards
+    if (!decodedTxn.isLegacy()) {
+      throw new DeviceCompatibilityError(
+        DeviceCompatibilityErrorType.INVALID_SDK_OPERATION,
+      );
+    }
+  }
+
```

Comment:
Create a helper function to check feature compatibility.

```suggestion
   if (!decodedTxn.isLegacy()) {
     await checkFeatureCompatibility(sdk, APP_SUPPORT_EIP1559);
   }
```